### PR TITLE
[Feature] Updates FTPHook provider to have test_connection

### LINKED ISSUE #21996: Add test_connection to FTP Hook
### Description

I would like to test if FTP connections are properly setup.

### Use case/motivation

To test FTP connections via the Airflow UI similar to https://github.com/apache/airflow/pull/19609

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by frankcash**
## Description

- Adds `test_connection` to `FTPHook`
- Adds related unit test
- Updates comment in `SFTPHook.test_connection`

closes: https://github.com/apache/airflow/issues/21996
Related: https://github.com/apache/airflow/pull/19609

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by josh-fell** [comment]
It would be a good idea to add a `test_connection_failed` test as well.

Also, it doesn't need to be in this PR but feel free to add if you feel up to it, to also add tests for `test_connection()` for the `FTPSHook` too especially since it inherits from `FTPHook` and overwrites `get_conn()` (which is used in `FTPHook.test_connection()`). WDYT?

**Comment by frankcash** [comment]
@josh-fell thanks for the feedback, I will definitely create a situation where it would return connection failed.  

As far as testing the `FTPSHook()` I think is a bit out of scope of what I aimed to do

**Review by uranusjr — APPROVED** [review]

File: tests/providers/ftp/hooks/test_ftp.py [hunk 1]
```diff
@@ -120,6 +120,20 @@ def test_retrieve_file_with_callback(self):
...
+            assert msg == 'Connection successfully tested'
+        
+    def test_connection_failure(self):
+        self.conn_mock = mock.MagicMock(name='conn_mock', side_effect=Exception('Test'))
+        fh.FTPHook.get_conn = self.conn_mock
+        with fh.FTPHook() as ftp_hook:
+            status, msg = ftp_hook.test_connection()
+            assert status is False
+            assert msg == 'Test'
+
```
Comment (eladkal) [inline tests/providers/ftp/hooks/test_ftp.py:136]:
```suggestion
```

to fix static checks

File: tests/providers/ftp/hooks/test_ftp.py [hunk 2]
```diff
@@ -120,6 +120,19 @@ def test_retrieve_file_with_callback(self):
...
+            assert status is True
+            assert msg == 'Connection successfully tested'
+        
+    def test_connection_failure(self):
+        self.conn_mock = mock.MagicMock(name='conn_mock', side_effect=Exception('Test'))
+        fh.FTPHook.get_conn = self.conn_mock
+        with fh.FTPHook() as ftp_hook:
+            status, msg = ftp_hook.test_connection()
+            assert status is False
+            assert msg == 'Test'
```
Comment (eladkal) [inline tests/providers/ftp/hooks/test_ftp.py]:
```suggestion
            assert msg == 'Test'

```

to fix the static checks

File: tests/providers/ftp/hooks/test_ftp.py [hunk 3]
```diff
@@ -120,6 +120,19 @@ def test_retrieve_file_with_callback(self):
             ftp_hook.retrieve_file(self.path, _buffer, callback=func)
         self.conn_mock.retrbinary.assert_called_once_with('RETR path', func)
 
+    def test_connection_success(self):
+        with fh.FTPHook() as ftp_hook:
+            status, msg = ftp_hook.test_connection()
+            assert status is True
+            assert msg == 'Connection successfully tested'
+
```
Comment (eladkal) [inline tests/providers/ftp/hooks/test_ftp.py]:
```suggestion

```

File: tests/providers/ftp/hooks/test_ftp.py (hunk 2 above)
Comment (frankcash) [reply tests/providers/ftp/hooks/test_ftp.py]: `Suggestion cannot be identical to original text. `

**Comment by frankcash** [comment]
@eladkal I merged the changes I could, but it appears to still be failing the static checks

**Comment by eladkal** [comment]
I rebased and applied a fix to the static checks.. lets see if it helps

**Comment by frankcash** [comment]
Thanks @eladkal

**Comment by potiuk** [comment]
A little mishap with this one as it broke the main tests (but it is no-one's fault :) #22026 fixes it and prevents similar problems like that in the future)

**Comment by eladkal** [comment]
> A little mishap with this one as it broke the main tests (but it is no-one's fault :) #22026 fixes it and prevents similar problems like that in the future)

ahhhh

**Pull request merged**